pkg/features/galley/galley.go

```diff
@@ -0,0 +1,29 @@
+//  Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package galley provides access to environment variables that enable/disable
+// Galley's features.
+package galley
+
+import (
+	"os"
+	"strconv"
+)
+
+var ConvertK8SService = parseBool(os.Getenv("ISTIO_CONVERT_K8S_SERVICE"))
+
+func parseBool(s string) bool {
```

Comment:
These should be placed within Galley sourcebase. They shouldn't be in a common folder.

Reply:
Funny, I was asked to "Move this to pkg/features/galley/galley.go similar to how it's done in pilot.". Why is similar pilot's stuff there in such case?

Should I just revert https://github.com/istio/istio/pull/10427/commits/43e61c608d8adbe5f3683fddc20233e02d491937 ?

Reply:
No, we should just have the code under galley/ and make sure that it is plumbed through option objects from the entry point, rather than have a direct check on a static.

Reply:
Pilot code being there is wrong IMO. We should move it out. Pilot specific code should live under pilot folder. Same for galley or mixer.